File: macos/controllers/preference-view-controllers/MUConditionsPreferencesViewController.m
//
// MUConditionsPreferencesViewController.m
//
// Copyright (c) 2014 3James Software. All rights reserved.
//

#import "MUConditionsPreferencesViewController.h"

@implementation MUConditionsPreferencesViewController

@synthesize identifier = _identifier;
@synthesize toolbarItemImage = _toolbarItemImage;
@synthesize toolbarItemLabel = _toolbarItemLabel;

- (instancetype) init
{
  if (!(self = [super initWithNibName: @"MUConditionsPreferencesView" bundle: nil]))
    return nil;

  _identifier = @"conditions";
  _toolbarItemImage = nil;
  _toolbarItemLabel = _(MULPreferencesConditions);

  NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];

  _conditionsArray = [NSKeyedUnarchiver unarchiveObjectWithData: [defaults objectForKey: MUPConditions]];

  return self;
}

@end

File: macos/controllers/preference-view-controllers/MUConditionsPreferencesViewController.swift
//
//  MUConditionsPreferencesViewController.swift
//  Koan
//
//  Created by Tyler Berry on 5/19/18.
//  Copyright © 2018 3James Software. All rights reserved.
//

import Cocoa

class MUConditionsPreferencesViewController: NSViewController, MASPreferencesViewController
{
  override var nibName: String? { return "MUConditionsPreferencesView" }
  override var identifier: String?
  {
    get { return _identifier }
    set { _identifier = newValue }
  }
  
  var _identifier: String? = "conditions"
  var toolbarItemImage: NSImage { return NSImage (named: NSImageNameCaution)! }
  var toolbarItemLabel: String { return NSLocalizedString (MULPreferencesConditions, comment: "") }
  
  var conditions = UserDefaults.standard.array (forKey: MUPConditions)
}
